includes/internal/emails/views/html-settings.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * File containing view for the tools page.
+ *
+ * @package sensei
+ * @since $$next-version$$
+ *
+ * @var array              $options Array of Sensei settings and corresponding values.
+ * @var array              $fields_to_display Array of Email settings to display.
+ * @var Email_Settings_Tab $this Email_Settings_Tab instance.
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$fields_to_display = isset( $fields_to_display ) ? $fields_to_display : [];
+$options           = isset( $options ) ? $options : [];
+?>
+<form id="email-notification-settings-form" action="options.php" method="post">
+	<?php settings_fields( 'sensei-settings' ); ?>
+	<?php
+	foreach ( $options as $option_name => $option_data ) {
+		// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
+		$this->form_field_hidden( $option_name, $option_data );
+	}
+	?>
+	<table class="form-table">
+		<?php foreach ( (array) $fields_to_display as $field ) { ?>
+			<tr<?php echo $field['args']['class']; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
+				<?php if ( ! empty( $field['args']['label_for'] ) ) { ?>
+					<th scope="row">
+						<label for="<?php echo esc_attr( $field['args']['label_for'] ); ?>">
+							<?php echo $field['title']; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+						</label>
+					</th>
+				<?php } else { ?>
+					<th scope="row">
+						<?php echo $field['title']; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
```

Comment:
Instead of the comments, should we try escaping here instead of earlier? Not something pressing and it also works as expected, but probably it's better to escape late if possible

Reply:
The output of this table is almost a copy of [the `do_settings_fields` function](https://developer.wordpress.org/reference/functions/do_settings_fields/).
They don't escape it anyhow. I thought they have reasons for that :)
Honestly, I can't find any default escaping for title in WP code during registration/adding fields, so I'm going to add mine here as you suggested. Also, we know what options we're going to display and it might become "broken" only in case of external modification.
Thank you!